programs/cleos/do_http_post_libcurl.cpp

```diff
@@ -0,0 +1,135 @@
+
+
+#include "do_http_post.hpp"
+#include <curl/curl.h>
+#include <eosio/chain/exceptions.hpp>
+
+namespace eosio { namespace client { namespace http {
+
+   static size_t write_callback(void* data, size_t size, size_t nmemb, void* userp) {
+      size_t       realsize = size * nmemb;
+      std::string& mem      = *(std::string*)userp;
+      mem.append((const char*)data, realsize);
+      return realsize;
+   }
+
+   void dump(const char* text, FILE* stream, unsigned char* ptr, size_t size) {
+      size_t       i;
+      size_t       c;
+      unsigned int width = 0x10;
+
+      fprintf(stream, "%s, %10.10ld bytes (0x%8.8lx)\n", text, (long)size, (long)size);
+
+      for (i = 0; i < size; i += width) {
+         fprintf(stream, "%4.4lx: ", (long)i);
+
+         /* show hex to the left */
+         for (c = 0; c < width; c++) {
+            if (i + c < size)
+               fprintf(stream, "%02x ", ptr[i + c]);
+            else
+               fputs("   ", stream);
+         }
+
+         /* show data on the right */
+         for (c = 0; (c < width) && (i + c < size); c++) {
+            char x = (ptr[i + c] >= 0x20 && ptr[i + c] < 0x80) ? ptr[i + c] : '.';
+            fputc(x, stream);
+         }
+
+         fputc('\n', stream); /* newline */
+      }
+   }
+
+   int my_trace(CURL* handle, curl_infotype type, char* data, size_t size, void* userp) {
```

Comment:
Seems like some of this code was likely copied from somewhere. Do we need to attribute the source?